Formula/file-roller.rb

```diff
@@ -22,18 +22,40 @@ class FileRoller < Formula
   depends_on "hicolor-icon-theme"
   depends_on "gnome-icon-theme"
 
+  # Add undeclared dependencies reported by `brew linkage`
+  depends_on "atk"
+  depends_on "cairo"
+  depends_on "gdk-pixbuf"
+  depends_on "gettext"
+  depends_on "glib"
+  depends_on "pango"
+
+  # These can be removed when the next version (>3.16.4) is released, since they
+  # are only required to support the patch below, which will no longer be needed
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "yelp-tools" => :build
+
+  # Remove when >3.16.4 is released
```

Comment:
I can see this is useful not only for patch of certain install method, but for every possible DSL in the formulae(like option, dependency, resources, caveats, test etc). However, it would be hard to declare similar DSL for all of them. So may I suggest a different approach? 

Instead of leveraging ruby DSL, why not use Ruby comment. For example:
```ruby
# VERSION 1.0.0 ONLY 
patch do
  ...
end

# VERSION 1.0.0 ONLY 
depends_on "foo"
```

Then we can use simply grep in `brew audit` to check this. The benefit is we don't need to introduce new DSL. And Ruby comment can pretty much be applied to every possible fields.